controllers/glanceapi_controller.go

```diff
@@ -566,7 +567,45 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 	// run check OpenStack secret - end
 
-	err = r.generateServiceConfig(ctx, helper, instance, &configVars)
+	// Get Enabled backends from customServiceConfig and run pre backend conditions
+	availableBackends := glancev1.GetEnabledBackends(instance.Spec.CustomServiceConfig)
+	_, hashChanged, err := r.createHashOfBackendConfig(instance, availableBackends)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Update the current StateFulSet (by recreating it) only when a backend is
+	// added or removed from an already existing API
+	if hashChanged {
+		if err = r.glanceAPIRefresh(ctx, helper, instance); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	// iterate over availableBackends for backend specific cases
+	for i := 0; i < len(availableBackends); i++ {
+		backendToken := strings.SplitN(availableBackends[i], ":", 2)
+		switch {
+		case backendToken[1] == "cinder":
+			cinder := &cinderv1.Cinder{}
+			err := r.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: glance.CinderName}, cinder)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// Request object not found, can't run GlanceAPI with this config
+					r.Log.Info("Cinder resource not found. Waiting for it to be deployed")
+					return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+				}
```

Comment:
I know you didn't add this block in this PR (it was moved from elsewhere), but one thought for a follow-up: maybe this message should go into a condition and thus be surfaced in the status so that a user can see it on the CLI?

Reply:
Yeah, it's actually a good idea, I'll prepare a follow up w/ that particular change, thank you @abays for the suggestion!